### PR TITLE
Add the MUL reference kernel to Tensorflow Lite Micro

### PR DESCRIPTION
**Pull request by jenselofsson**
This PR adds support for the mul reference kernel to Tensorflow Lite Micro. It supports float and uint8. Broadcast support is not included. The testcases included is copied from the testcases in Tensorflow Lite.

### CODE REVIEW

File: tensorflow/lite/experimental/micro/kernels/mul_test.cc [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+          F2Q(-0.8, input_min, input_max), F2Q(0.2, input_min, input_max),
+          F2Q(0.9, input_min, input_max), F2Q(0.7, input_min, input_max),
+      },                // input1 data
+      {4, 1, 2, 2, 1},  // input2 dims
+      {
+          F2Q(0.6, input_min, input_max), F2Q(0.4, input_min, input_max),
+          F2Q(0.9, input_min, input_max), F2Q(0.8, input_min, input_max),
+      },  // input2 data
+      input_min,
+      input_max, {4, 1, 2, 1, 2},  // output dims
```
Comment (jenselofsson) [inline tensorflow/lite/experimental/micro/kernels/mul_test.cc]: The shape should be {4, 1, 2, 2, 1}.

File: tensorflow/lite/experimental/micro/kernels/mul.cc [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+==============================================================================*/
+
+#include "tensorflow/lite/kernels/internal/reference/mul.h"
+
+#include "tensorflow/lite/c/c_api_internal.h"
+#include "tensorflow/lite/kernels/internal/quantization_util.h"
+#include "tensorflow/lite/kernels/internal/tensor.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+
+#include <stdio.h>
```
Comment (jenselofsson) [inline tensorflow/lite/experimental/micro/kernels/mul.cc]: This should be removed.

**Comment by gbaned** [comment]
@jenselofsson Could you please resolve the conflicts? Thanks!

File: tensorflow/lite/experimental/micro/kernels/mul_test.cc [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+  node.custom_initial_data_size = 0;
+  node.delegate = nullptr;
+
+  if (registration->prepare) {
+    TF_LITE_MICRO_EXPECT_EQ(kTfLiteOk, registration->prepare(&context, &node));
+  }
+
+  TF_LITE_MICRO_EXPECT_NE(nullptr, registration->invoke);
+  TF_LITE_MICRO_EXPECT_EQ(kTfLiteOk, registration->invoke(&context, &node));
+}
```
Comment (jenselofsson) [inline tensorflow/lite/experimental/micro/kernels/mul_test.cc]: These functions needs to verify that the output data actually is what's expected. Same for TestMulQuantized

**Comment by jenselofsson** [comment]
I'm working on getting int8 support, so I'll open up a new PR with support for both int8 and uint8 when it's finished.

**Pull request closed without merging**